packages/config/lib/index.js

```diff
@@ -20,6 +20,7 @@ const breakingKeys = _.map(breakingOptions, 'name')
 const defaultValues = createIndex(options, 'name', 'defaultValue')
 const publicConfigKeys = _(options).reject({ isInternal: true }).map('name').value()
 const validationRules = createIndex(options, 'name', 'validation')
+const writeableOptions = options.filter((option) => option.isWriteable).map((option) => option.name)
```

Comment:
Let's reduce or [`_.transform`](https://lodash.com/docs/4.17.15#transform) this into an object, then below `validateNoReadOnlyConfig` can use `writeableOptions[element]` and make the lookup O(1).